From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ayu Kashyap <ayushkumar274549@gmail.com>
Date: Thu, 23 Oct 2025 03:00:00 +0530
Subject: [PATCH] Haptics: Use brightness file for kl8h

Change haptics logic to use /sys/class/leds/vibrator/brightness
as this is the correct path for this device.
Increase haptics duration to 200ms.

Change-Id: I8b06ac64a1c7f2bb79a1295c505950afbf46a3d3
---
 minuitwrp/events.cpp | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/minuitwrp/events.cpp b/minuitwrp/events.cpp
index 8a34f0e..b8a3e2a 100644
--- a/minuitwrp/events.cpp
+++ b/minuitwrp/events.cpp
@@ -192,10 +192,17 @@
         write_to_file(VIBRATOR_TIMEOUT_FILE, tout);
     }
 #else
-    if (std::ifstream(LEDS_HAPTICS_ACTIVATE_FILE).good()) {
-        write_to_file(LEDS_HAPTICS_DURATION_FILE, tout);
-        write_to_file(LEDS_HAPTICS_ACTIVATE_FILE, "1");
+    /* Custom Haptics Logic: Use brightness file */
+    if (std::ifstream("/sys/class/leds/vibrator/brightness").good()) {
+        write_to_file("/sys/class/leds/vibrator/brightness", "255");
+        usleep(200 * 1000); // 200ms
+        write_to_file("/sys/class/leds/vibrator/brightness", "0");
+    } else if (std::ifstream(LEDS_HAPTICS_ACTIVATE_FILE).good()) {
+        write_to_file(LEDS_HAPTICS_DURATION_FILE, tout);
+        write_to_file(LEDS_HAPTICS_ACTIVATE_FILE, "1");
     } else
         write_to_file(VIBRATOR_TIMEOUT_FILE, tout);
 #endif
     return 0;
 }
 #endif

-- 
2.39.2